Essentials/src/main/java/com/earth2me/essentials/Jails.java

```diff
@@ -318,5 +353,9 @@ public void onJailPlayerJoin(final PlayerJoinEvent event) {
                 future.completeExceptionally(ex);
             }
         }
+
+        private boolean shouldIgnore(final Player base) {
+            return base.getUniqueId().version() <= 2;
```

Comment:
Better detection for NPCs specifically. Previous check also triggers for Geyser users which I would prefer we don't intentionally break (even if we do not officially support it).
```suggestion
            return base.getUniqueId().version() == 2 || base.hasMetadata("NPC");
```